packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -82,6 +82,17 @@ void tryDelete(String dependencyBaseName, Directory globalCachePub, Logger logge
   logger.printWarning('The join of pub-caches failed');
 }
 
+void tryToDeleteLocalCache(Directory localCache, Logger logger) {
+  try {
+    if (localCache.existsSync()) {
+      localCache.deleteSync(recursive: true);
+    }
+  }
+  on FileSystemException catch(ex) {
```

Comment:
Here and elsewhere the `on ...` clause should be on the same line as the previous close curly brace